Do not force static check

On distros such as Arch installing check might not install the static library. Building on those distros will fail because of that.

Yes there was a build error.

After `sudo pacman -S gperf check`, then run `mkdir build && cd build && cmake -GNinja ..` (using Makefile gives the same result) and:

```
$ cmake -GNinja ..
-- The C compiler identification is GNU 10.2.0
-- The CXX compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3") 
-- Found LibUSB1: usb-1.0 (found version "1.0.23") 
-- Found LibFTDI1: ftdi1;usb-1.0 (found version "1.4") 
-- Found LibZip: zip (found suitable version "1.6.1", minimum required is "1.0") 
CMake Error at /usr/lib64/cmake/check/check-targets.cmake:96 (message):
  The imported target "Check::check" references the file

     "/usr/lib/libcheck.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/cmake/check/check-targets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/cmake/check/check-config.cmake:36 (include)
  CMakeLists.txt:15 (find_package)


-- Configuring incomplete, errors occurred!
```

The cmake library for check seems to be shipped with the `check` package. Currently the packaged version is `0.15.2`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,13 +13,13 @@ find_package(LibUSB1 REQUIRED)
 find_package(LibFTDI1 REQUIRED)
 find_package(LibZip 1.0 REQUIRED)
 find_package(Check CONFIG NAMES Check check)
-if (NOT TARGET Check::check)
+if (NOT TARGET Check::checkShared)
 	find_package(PkgConfig REQUIRED)
 
 	pkg_search_module(CHECK REQUIRED Check check)
 
-	add_library(Check::check INTERFACE IMPORTED)
-	set_target_properties(Check::check PROPERTIES
+	add_library(Check::checkShared INTERFACE IMPORTED)
+	set_target_properties(Check::checkShared PROPERTIES
 		INTERFACE_LINK_LIBRARIES "${CHECK_LDFLAGS}"
 		INTERFACE_COMPILE_OPTIONS "${CHECK_CFLAGS}")
 endif()
@@ -131,7 +131,7 @@ foreach(test_source IN ITEMS ${TESTS})
 	string(REPLACE / _ test_name ${test_name})
 	string(REPLACE .c "" test_name ${test_name})
 	add_executable(${test_name} ${test_source})
-	target_link_libraries(${test_name} Check::check openvizsla_static)
+	target_link_libraries(${test_name} Check::checkShared openvizsla_static)
 	target_compile_definitions(${test_name} PRIVATE PROJECT_ROOT="${PROJECT_SOURCE_DIR}")
 	add_test(${test_name} ${test_name})
 endforeach(test_source)

